src/SqlAsyncEnumerable.cs

```diff
@@ -101,23 +100,22 @@ public ValueTask<bool> MoveNextAsync()
             /// </returns>
             private async Task<bool> GetNextRowAsync()
             {
-                if (this._reader == null)
+                if (this._connection.State != System.Data.ConnectionState.Closed)
```

Comment:
Assuming the tests pass please create an issue for investigating this and then add a comment explaining why we're doing this with a link to the issue.

Reply:
I ran locally and they passed (fingers crossed), will add the comment, thanks